packages/workbox-routing/_default.mjs

```diff
@@ -161,14 +161,27 @@ class DefaultRouter extends Router {
     }
 
     const cacheName = cacheNames.getPrecacheName(options.cacheName);
-    const handler = () => caches.match(cachedAssetUrl, {cacheName});
+    const handler = () => caches.match(cachedAssetUrl, {cacheName})
+      .then((response) => {
+        if (response) {
+          return response;
+        }
+        throw new Error(`The cache ${cacheName} did not have an entry for ` +
+          `${cachedAssetUrl}.`);
+      }).catch((error) => {
+        if (process.env.NODE_ENV !== 'production') {
+          logger.debug(`Unable to respond to navigation request with cached ` +
+            `response: ${error}. Falling back to network.`);
```

Comment:
This should be `error.message` or you should log it as a separate log call (possibly in a group).